internal/client/blueprints_test.go

```diff
@@ -1229,3 +1229,48 @@ func TestBlueprintChangeV1(t *testing.T) {
 	require.NotNil(t, bp, "GET blueprint change failed: missing blueprint")
 	assert.Equal(t, bp.Version, "0.0.1")
 }
+
+func TestEmptyPackageNameBlueprintJsonV0(t *testing.T) {
+	for _, packageSnippet := range []string{
+		`"packages": [{"name": "", "version": "*"}]`,
+		`"packages": [{"name": ""}]`,
+	} {
+		bp := `{
+		"name": "test-emptypackage-blueprint-v0",
+		"description": "TestEmptyPackageNameBlueprintV0",
+		"version": "0.0.1",
+        ` + packageSnippet + `}`
+
+		expectedErrorPrefix := "BlueprintsError: All package entries need to contain the name of the package."
+
+		resp, err := PostJSONBlueprintV0(testState.socket, bp)
+		require.NoError(t, err, "failed with a client error")
+		require.False(t, resp.Status, "Negative status expected.")
+		require.Equal(t, len(resp.Errors), 1, "There should be exactly one error")
+		reasoning := "I expected an error message starting with\n" + expectedErrorPrefix +
+			"\nnot\n" + resp.Errors[0].String()
+		assert.True(t, strings.HasPrefix(resp.Errors[0].String(), expectedErrorPrefix), reasoning)
```

Comment:
Related to the above, the `assert.Equal()` output is really nice, so if we do the above we also get nice diffs in case of failure :)